src/geo/map.js

```diff
@@ -87,6 +88,24 @@ var Map = Model.extend({
 
   // PUBLIC API METHODS
 
+  addError: function (error) {
```

Comment:
Have you thought how error events are going to be transmitted to other components (Builder, DI...)?

Reply:
I thought about doing something like:

```
model.errors.on('add', function () {
  // Do something here
});
```
in the map model in deep insights integration, but to be honest I don't know if that's the best approach...

Reply:
How about trigger an `error` event from the model with the error type?

Reply:
So, trigger an error Instead of having a collection of errors?

Reply:
no, whenever an error is added to the collection, instead of listening the `add` event's collection, we could trigger an event from the map model itself, with the error type. This way we can avoid finding the error type in the consumer. Something like:
```
model.trigger('error', {type: error.type});
or 
model.trigger('error:limits') (with namespace)
```

Reply:
Just an idea, we need to think what happens when the error is gone, could this happen in the middle of a session?

Reply:
+1 for the idea of trigger the error with the type.

About the error gone, I'm not sure, in the limits case, how do we know the error is not happening anymore?

Reply:
> Just an idea, we need to think what happens when the error is gone, could this happen in the middle of a session?

Is this something we need to worry about?